hack/ocp_libvirt_setup.yaml

```diff
@@ -0,0 +1,218 @@
+        content: |
+          polkit.addRule(function(action, subject) {
+            if (action.id == "org.libvirt.unix.manage" && subject.local && subject.active && subject.isInGroup("wheel")) {
+                return polkit.Result.YES;
+            }
+          });
+        dest: /etc/polkit-1/rules.d/80-libvirt.rules
+        owner: root 
+        group: root
+        mode: 0644
+
+    - name: Setup IP Forwarding
+      sysctl:
+        name: net.ipv4.ip_forward
+        value: 1
+        sysctl_set: yes
+        state: present
+        reload: yes
+
+    - block:
+      - name: Ensure Libvirt is Configured for ... 
+        ini_file:
+          path: /etc/libvirt/libvirtd.conf
+          section: 
+          option: "{{ item.name }}" 
+          value: "{{ item.val }}" 
+          mode: 0644
+          owner: root
+          backup: yes
+        loop: "{{ libvirt_settings }}"
+
+      - name: Ensure Libvirt is Configured to LISTEN
+        ini_file:
+          path: /etc/sysconfig/libvirtd
+          section: 
+          option: "LIBVIRTD_ARGS" 
+          value: "--listen" 
+          mode: 0644
+          owner: root
+          backup: yes
+        when: ansible_distribution == 'Fedora'
+
+      - fail:
+          msg: "This was designed to work on Fedora (please file a PR)"
+        when: ansible_distribution != 'Fedora'
+
+    - name: "Restart and Enable libvirtd" 
+      systemd:
+        name: "libvirtd"
+        state: restarted
+        enabled: yes
+        daemon_reload: yes
+
+    become: yes
+
+  ### Make sure Libvirt Networks don't conflict with 192.168.126.0/24
+  - block: 
+    - name: Get Libvirt Networks
+      virt_net:
+        command: list_nets
+      register: libvirt_networks
+
+    - debug:
+        var: libvirt_networks.list_nets
+
+    - name: Get Libvirt Network XML
+      virt_net: 
+        command: get_xml
+        name: "{{ item }}" 
+      loop: "{{ libvirt_networks.list_nets }}"
+      register: libvirt_networks_xml
+
+    - name: Get data
+      xml:
+        xmlstring: "{{ item.get_xml }}"
+        xpath: /network/ip
+        content: attribute
+        attribute: address
+      register: ip_address
+      loop: "{{ libvirt_networks_xml.results }}"
+
+    - set_fact:
+        net_masks: "{{ item.matches.0.ip.address.split('.')[0:3] | join('.')}}.0/{{ item.matches.0.ip.netmask }}"
+      loop: "{{ ip_address.results }}" 
+      register: libvirt_netmasks
+
+    - fail:
+        msg: "Net Mask {{ item }} confilcts with {{ cluster.cluster_cidr }}"
+      when: item == cluster.0.cluster_cidr 
+      loop: "{{ libvirt_netmasks.results | map(attribute='ansible_facts.net_masks') | list | ipaddr('net') }}"
+
+    become: yes
+
+  ### Confirm that Storage Pool is defined
+  - block:     ### This should fail here if the pool is not defined 
+    - virt_pool:
+        command: status
+        name: "default"
+      register: storage_pool
+
+  - block: 
+    - name: Enable dnsmasq dns for NetworkManager
```

Comment:
It would be nice if we are adding automation to carry out some of the libvirt setup for us to also have automation to undo it. The unfortunate truth of the libvirt method is that the changes being made to the system are not trivial.